src/internal/connector/onedrive/drive.go

```diff
@@ -143,9 +143,34 @@ func collectItems(
 	// https://docs.microsoft.com/en-us/graph/api/driveitem-delta?
 	// view=graph-rest-1.0&tabs=http#example-4-retrieving-delta-results-using-a-timestamp
 	builder := service.Client().DrivesById(driveID).Root().Delta()
+	pageCount := int32(999)    // max we can do is 999
+	requestFields := []string{ // TODO(meain): shrink this list to only needed ones
```

Comment:
Per TODO: I'd look at what we're persisting in the onedrive and sharepoint item info, and stick with that.